tests/expectations.json

```diff
@@ -8573,86 +8572,7 @@
-  "built-ins/TypedArray/prototype/map/BigInt/speciesctor-get-species-custom-ctor-length-throws-resizable-arraybuffer.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/BigInt/speciesctor-get-species-custom-ctor-length-throws.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/BigInt/speciesctor-get-species-custom-ctor-length.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/BigInt/speciesctor-get-species-custom-ctor-returns-another-instance.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/BigInt/speciesctor-get-species-custom-ctor-throws.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/BigInt/speciesctor-get-species-custom-ctor.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/BigInt/speciesctor-get-species-returns-throws.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/BigInt/speciesctor-get-species-use-default-ctor.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/BigInt/speciesctor-get-species.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/BigInt/values-are-not-cached.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/arraylength-internal.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/callbackfn-arguments-with-thisarg.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/callbackfn-arguments-without-thisarg.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/callbackfn-detachbuffer.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/callbackfn-is-not-callable.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/callbackfn-no-interaction-over-non-integer-properties.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/callbackfn-not-called-on-empty.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/callbackfn-resize.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/callbackfn-return-affects-returned-object.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/callbackfn-return-does-not-change-instance.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/callbackfn-return-does-not-copy-non-integer-properties.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/callbackfn-returns-abrupt.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/callbackfn-set-value-during-interaction.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/callbackfn-this.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/detached-buffer.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/invoked-as-func.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/invoked-as-method.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/resizable-buffer-grow-mid-iteration.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/resizable-buffer-shrink-mid-iteration.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/resizable-buffer.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/return-abrupt-from-this-out-of-bounds.js": "FAIL",
-  "built-ins/TypedArray/prototype/map/return-new-typedarray-conversion-operation-consistent-nan.js": "FAIL",
   "built-ins/TypedArray/prototype/map/return-new-typedarray-conversion-operation.js": "FAIL",
```

Comment:
It seems to be failing during comparison.

```
thread 'main' panicked at nova_vm/src/ecmascript/types/language/number.rs:542:17:
assertion failed: y.into_f64().to_bits() == (-0.0f64).to_bits() ||
    (x.into_i64() as f64) != y.into_f64()
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
Alright, I fixed this by separating SmallBigInt from SmallInteger entirely.